src/program/l2fwd/l2fwd.lua

```diff
@@ -0,0 +1,70 @@
+module(...,package.seeall)
+
+local lib = require("core.lib")
+local main = require("core.main")
+
+L2Fwd = {}
+
+function L2Fwd.new(conf)
+   local o = {}
+   return setmetatable(o, { __index = L2Fwd })
+end
+
+function L2Fwd:push()
+   local input, output = assert(self.input.input), assert(self.output.output)
+
+   while not link.empty(input) and not link.full(output) do
+      link.transmit(output, link.receive(input))
+   end
+end
+
+local function show_usage(code)
+   print(require("program.l2fwd.README_inc"))
+   main.exit(code)
+end
+
+local function parse_args(args)
+   local handlers = {}
+   local opts = { duration = nil }
+   function handlers.h() show_usage(0) end
+   function handlers.t() opts.use_tap = true end
+   function handlers.v() opts.use_virtionet = true end
```

Comment:
I think there is a syntax for allowing the drivers to be specified on the command line in a textual way, like `virtio:...`.  I could be wrong though.  However that would be better than command-line options and it would even allow for forwarding from a NIC to vhost-user, for example.